runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -24,51 +26,82 @@
  */
 package net.runelite.client.plugins.mining;
 
+import com.google.inject.Provides;
 import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import javax.inject.Inject;
 import lombok.AccessLevel;
 import lombok.Getter;
+import lombok.Setter;
 import lombok.extern.slf4j.Slf4j;
 import net.runelite.api.Client;
 import net.runelite.api.GameObject;
 import net.runelite.api.GameState;
+import net.runelite.api.InventoryID;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.ItemID;
 import static net.runelite.api.ObjectID.DEPLETED_VEIN_26665;
 import static net.runelite.api.ObjectID.DEPLETED_VEIN_26666;
 import static net.runelite.api.ObjectID.DEPLETED_VEIN_26667;
 import static net.runelite.api.ObjectID.DEPLETED_VEIN_26668;
 import static net.runelite.api.ObjectID.EMPTY_WALL;
 import net.runelite.api.WallObject;
+import net.runelite.api.events.ChatMessage;
 import net.runelite.api.events.GameObjectDespawned;
 import net.runelite.api.events.GameStateChanged;
 import net.runelite.api.events.GameTick;
+import net.runelite.api.events.MenuOptionClicked;
 import net.runelite.api.events.WallObjectSpawned;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.config.ConfigManager;
 import net.runelite.client.eventbus.Subscribe;
 import net.runelite.client.plugins.Plugin;
 import net.runelite.client.plugins.PluginDescriptor;
 import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.util.Text;
 
 @Slf4j
 @PluginDescriptor(
-	name = "Mining",
-	description = "Show ore respawn timers",
-	tags = {"overlay", "skilling", "timers"},
-	enabledByDefault = false
+		name = "Mining",
+		description = "Show ore respawn timers and coal bag overlay",
+		tags = {"overlay", "skilling", "timers", "coal", "coalbag", "coal bag"},
+		enabledByDefault = false
 )
 public class MiningPlugin extends Plugin
 {
 	private static final int ROCK_DISTANCE = 14;
 	private static final int MINING_GUILD_REGION = 12183;
 
+	private static final Pattern COAL_BAG_EMPTY_MESSAGE = Pattern.compile("^The coal bag is (now )?empty\\.$");
+	private static final Pattern COAL_BAG_ONE_MESSAGE = Pattern.compile("^The coal bag contains one piece of coal\\.$");
+	private static final Pattern COAL_BAG_AMOUNT_MESSAGE = Pattern.compile("^The coal bag contains (\\d+) pieces of coal\\.$");
```

Comment:
This and `COAL_BAG_ONE_MESSAGE` could be combined

Reply:
Combining the two regexes would mean the check later would require going through the regex groups to check whether it is ONE or AMOUNT. I think this solution is more readable, but correct me if there's a simpler way.